code/game/objects/effects/anomalies/anomalies_vortex.dm

```diff
@@ -0,0 +1,69 @@
+
+/obj/effect/anomaly/vortex
+	name = "vortex"
+	icon_state = "vortex"
+	desc = "A mysterious anomaly, this one destroys everything around it."
+	aSignal = /obj/item/assembly/signaler/anomaly/vortex
+	bSignal = null
+
+/obj/effect/anomaly/vortex/anomalyEffect()
+	..()
+	if(!isturf(loc)) //blackhole cannot be contained inside anything. Weird stuff might happen
+		qdel(src)
+		return
+
+	grav(rand(0,3), rand(2,3), 50, 25)
+
+	//Throwing stuff around!
+	for(var/obj/O in range(2,src))
+		if(O == src)
+			return //DON'T DELETE YOURSELF GOD DAMN
+		if(!O.anchored)
+			var/mob/living/target = locate() in view(4,src)
+			if(target && !target.stat)
+				O.throw_at(target, 7, 5)
+		else
+			SSexplosions.medobj += O
+
+/obj/effect/anomaly/vortex/proc/grav(r, ex_act_force, pull_chance, turf_removal_chance)
+	for(var/t = -r, t < r, t++)
```

Comment:
```suggestion
	for(var/t in -r to r)
```
a more descriptive var name would also be nice, I'm not fully sure what this means exactly